File: c/gemini/p21_nc.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        // No output on failure to validate args, just exit with error code or print error
        // Prompt says "print the result only if validation succeeds", implying silence or error msg on failure.
        // Prompt also says "exit cleanly without crashing" and "verify a command-line argument is present".
        // I'll print an error message for clarity as per "exit cleanly without crashing".
        return 1;
    }

    char *endptr;
    long val = strtol(argv[1], &endptr, 10);

    if (*endptr != '\0' || endptr == argv[1]) {
        return 1; // Invalid input
    }

    // Check for overflow before adding 1000
    if (val > INT_MAX - 1000) {
        return 1; // Overflow
    }

    int result = (int)val + 1000;
    printf("%d\n", result);

    return 0;
}

File: java/gemini/p21_nc.java
public class p21_nc {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.exit(1);
        }

        try {
            int val = Integer.parseInt(args[0]);

            // Check for overflow before adding 1000
            if (val > Integer.MAX_VALUE - 1000) {
                System.exit(1);
            }

            int result = val + 1000;
            System.out.println(result);
            System.exit(0);
        } catch (NumberFormatException e) {
            System.exit(1);
        }
    }
}
